misk/src/main/kotlin/misk/web/actions/ReadinessCheckService.kt

```diff
@@ -0,0 +1,83 @@
+package misk.web.actions
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.ServiceManager
+import misk.healthchecks.HealthCheck
+import misk.healthchecks.HealthStatus
+import misk.tasks.RepeatedTaskQueue
+import misk.tasks.Result
+import misk.tasks.Status
+import misk.web.ReadinessRefreshQueue
+import misk.web.WebConfig
+import wisp.logging.getLogger
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Singleton
+class ReadinessCheckService @Inject constructor(
+  private val config: WebConfig,
+  private val clock: Clock,
+  private val serviceManagerProvider: Provider<ServiceManager>,
+  @JvmSuppressWildcards private val healthChecks: List<HealthCheck>,
+  @ReadinessRefreshQueue private val taskQueue: RepeatedTaskQueue,
+): AbstractIdleService() {
+  var status: CachedStatus? = null
+    private set
+
+  init {
+    require(config.readiness_refresh_interval_ms < config.readiness_max_age_ms || config.health_port <= 0) {
```

Comment:
nit - should probably drop the `|| config.health_port <= 0` since that requirement applies either way